linkerd/stack/src/queue.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{
+    failfast::{self, FailFast},
+    layer::{self, Layer},
+    BoxService, CloneParam, ExtractParam, NewService, Param, Service,
+};
+use linkerd_error::Error;
+use std::{fmt, marker::PhantomData, time::Duration};
+use tower::buffer::Buffer;
+
+#[derive(Debug, Copy, Clone)]
+pub struct QueueConfig {
+    /// The number of requests (or connections, depending on the context) that
+    /// may be buffered
+    pub capacity: usize,
+
+    /// The maximum amount of time a request may be buffered before failfast
+    /// errors are emitted.
+    pub failfast_timeout: Duration,
+}
+
+pub struct NewQueue<N, Req, X = ()> {
```

Comment:
I generally prefer to keep the service types last (for type readability in debug output). I don't think the default parameter is really worth it in this case.